src/yunohost/backup.py

```diff
@@ -49,111 +51,199 @@
+
+        # Add unlisted files from backup tmp dir
+        self._mark_for_backup('backup.csv')
+        self._mark_for_backup('info.json')
+        if len(self.info['apps']) > 0:
+            self._mark_for_backup('apps')
+        if os.path.isdir(os.path.join(self.collect_dir, 'conf')):
+            self._mark_for_backup('conf')
+        if os.path.isdir(os.path.join(self.collect_dir, 'data')):
+            self._mark_for_backup('data')
+        self.csv_file.close()
+
+        # Calculate total size
+        self._compute_backup_size()
+
+        # Create backup info file
+        with open("%s/info.json" % self.collect_dir, 'w') as f:
+            f.write(json.dumps(self.info))
+
+    def backup(self, methods='tar', output_directory=None):
+        """
+        Apply backup methods
+
+        method: name of a method or list of names
+        """
+        if output_directory is None:
+            output_directory = ARCHIVES_PATH
+
+        if not isinstance(methods, basestring):
+            for method in methods:
+                self.backup(method, output_directory)
+            return
+        else:
+            method=methods
 
-    def _clean_tmp_dir(retcode=0):
-        ret = hook_callback('post_backup_create', args=[tmp_dir, retcode])
+        if method in ["copy", "tar", "borg"]:
+            logger.info(m18n.n('backup_applying_method_' + method))
+            getattr(self, "_" + method + "_files")(output_directory)
+            logger.info(m18n.n('backup_method_' + method + '_finished'))
+        else:
+            logger.info(m18n.n('backup_applying_method_custom', method=method))
+            self._hook_files(method, output_directory)
+            logger.info(m18n.n('backup_method_custom_finished', method=method))
+
+    def clean(self, retcode=-1):
+        """ Call post_backup_create hooks and delete collect_dir """
+        ret = hook_callback('post_backup_create', args=[self.collect_dir,
+                                                        retcode])
         if not ret['failed']:
-            filesystem.rm(tmp_dir, True, True)
+            if self._is_temp_collect_dir:
+                filesystem.rm(self.collect_dir, True, True)
             return True
         else:
             logger.warning(m18n.n('backup_cleaning_failed'))
             return False
 
-    # Create temporary directory
-    if not tmp_dir:
-        tmp_dir = "%s/tmp/%s" % (BACKUP_PATH, name)
-        if os.path.isdir(tmp_dir):
+    @property
+    def _is_temp_collect_dir(self):
+        return self.collect_dir == os.path.join(BACKUP_PATH, self.name)
+
+    def _define_backup_name(self):
+        """ Define backup name """
+        # FIXME: case where this name already exist
+        return time.strftime('%Y%m%d-%H%M%S')
+
+    def _init_collect_dir(self):
+        """ Initialize preparation directory """
+
+        if not self.collect_dir:
+            self.collect_dir = "%s/tmp/%s" % (BACKUP_PATH, name)
```

Comment:
It was dead code, but there was an error on the path the good one is BACKUP_PATH/tmp/name
Fixed with 27f6a979f8ba6596b66d2f7299ed600836c15ae1